src/parser/SpatialQuery.cpp

```diff
@@ -95,7 +115,7 @@ SpatialJoinConfiguration SpatialQuery::toSpatialJoinConfiguration() const {
         "spatialSearch: { [Config Triples] { <Something> <ThatSelects> ?right "
         "} }.");
   } else if (!ignoreMissingRightChild_ && !childGraphPattern_.has_value() &&
-             !payloadVariables_.empty()) {
+             !payloadVariables_.isAll() && !payloadVariables_.empty()) {
```

Comment:
Isn't the `!isAll` call redundant, because `isAll()` directly implies `!empty()` ?

Reply:
I disagree. If I'm not totally mistaken, this is neither logically nor semantically redundant.
- Logically because isAll implies not-empty but not-empty does not imply isAll
- Semantically because there is a subtle edge case, where this has an actual impact: Assume the user does a max-dist-join and declares left and right outside of the service. Then they give the configuration `<payload> <all>`. This is the default behavior anyway. But without the `!isAll()` here we would throw despite the user's configuration just stating the default value.